test/react_test_utils_test.dart

```diff
@@ -0,0 +1,277 @@
+import 'dart:js';
+
+import 'package:unittest/unittest.dart';
+import 'package:unittest/html_config.dart';
+import 'package:react/react.dart';
+import 'package:react/react_client.dart' as reactClient;
+import 'package:react/react_test_utils.dart';
+
+import 'test_components.dart';
+
+
+void main() {
+
+  useHtmlConfiguration();
+  reactClient.setClientConfiguration();
+
+  var component;
+  var domNode;
+  var _React = context['React'];
+  var _Object = context['Object'];
+
+  // This is intended only to make a simple representation of react elements
+  // as JS objects for testings purposes.
+  JsObject createReactElement(
+      String componentType, Map properties, List childrenComponents) {
+    var proto = new JsObject(_Object);
+    proto['render'] = () => null;
+    proto['setState'] = () => null;
+
+    var props = new JsObject(_Object);
+    properties.forEach((k, v) => props[k] = v);
+
+    var renderedChildren;
+    if (childrenComponents.length > 0) {
+      renderedChildren = new JsObject(_Object);
+      for (var i = 0; i < childrenComponents.length; i++) {
+        renderedChildren['.${i}'] = childrenComponents[i];
+      }
+    }
+
+    var component = new JsObject(_Object);
+    component['constructor'] = componentType.toLowerCase();
+    component['mountComponent'] = () => null;
+    component['props'] = props;
+    component['tagName'] = componentType.toUpperCase();
+    component['__proto__'] = proto;
+    component['_renderedChildren'] = renderedChildren;
+
+    return component;
+  };
```

Comment:
I think ``mockComponent()`` is supposed to do the same thing. It is the last thing I ported over and I did some reading online to see how to use it. Most people said they ended up writing their own as well because ``mockComponent()`` depends on <a href='https://groups.google.com/forum/#!topic/reactjs/u6CbOF3lIig' target='_blank'>Jest</a>.